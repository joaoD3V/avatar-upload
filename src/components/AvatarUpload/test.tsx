import 'jest-canvas-mock';
import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import { AvatarUpload } from '.';

describe('<AvatarUpload />', () => {
  it('should render without avatar by default', () => {
    const { container } = renderWithTheme(<AvatarUpload />);

    expect(screen.queryByLabelText(/close button/i)).not.toBeInTheDocument();
    expect(screen.getByLabelText(/wrapper/i)).toBeInTheDocument();
    expect(screen.queryByLabelText(/Avatar canva/i)).not.toBeInTheDocument();
    expect(
      screen.getByAltText(
        /Illustration of mountains and sun, symbolizing a picture icon/i
      )
    ).toBeInTheDocument();
    expect(screen.getByText(/Organization Logo/i)).toBeInTheDocument();
    expect(
      screen.getByText(/Drop the image here or click to browse./i)
    ).toBeInTheDocument();
    expect(screen.getByTestId(/drop-input/i)).toBeInTheDocument();
    expect(screen.getByTestId(/drop-wrapper/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with avatar', () => {
    const props = {
      src: 'https://99prod.s3.amazonaws.com/uploads/image/file/2429377/3eef648f6ce677d5e317509045df2be0.jpg',
      sliderValue: 0,
    };

    renderWithTheme(<AvatarUpload options={props} />);

    expect(screen.queryByLabelText(/Avatar canva/i)).toBeInTheDocument();
  });
});
