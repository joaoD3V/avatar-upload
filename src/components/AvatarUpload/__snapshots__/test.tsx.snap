// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvatarUpload /> should render without avatar by default 1`] = `
.c0 {
  width: 553px;
  height: 177px;
  background: #F2F5F8;
  border: 0;
  border-radius: 8px;
  padding: 32px;
  position: relative;
}

.c1 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  cursor: pointer;
}

.c1 canvas {
  pointer-events: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  font-weight: 500;
  font-size: 14px;
  color: #495567;
  margin-left: 12px;
}

.c5 {
  font-weight: 400;
  font-size: 14px;
  color: #677489;
  margin-top: 0.8rem;
}

@media (max-width:768px) {
  .c0 {
    width: 90vw;
    height: 277px;
    padding: 16px;
  }
}

@media (max-width:768px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
  }
}

<div
  aria-label="Wrapper"
  class="c0"
>
  <div
    class="c1"
    data-testid="drop-wrapper"
    role="button"
    tabindex="0"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <img
          alt="Illustration of mountains and sun, symbolizing a picture icon"
          class=""
          src="/img/image-icon.png"
        />
        <span
          class="c4"
        >
          Organization Logo
        </span>
      </div>
      <p
        class="c5"
      >
        Drop the image here or click to browse.
      </p>
    </div>
    <input
      accept="image/jpeg,image/png,image/webp"
      autocomplete="off"
      data-testid="drop-input"
      style="display: none;"
      tabindex="-1"
      type="file"
    />
  </div>
</div>
`;
